ivis/ivis.py

```diff
@@ -115,55 +115,57 @@ class than implements the collections.abc.Sequence class, specifically the
     """
 
     def __init__(self, embedding_dims=2, k=150, distance='pn', batch_size=128,
-                 epochs=1000, n_epochs_without_progress=20, ntrees=50,
+                 epochs=1000, n_epochs_without_progress=20, n_trees=50,
                  knn_distance_metric='angular', search_k=-1,
                  precompute=True, model='szubert',
                  supervision_metric='sparse_categorical_crossentropy',
                  supervision_weight=0.5, annoy_index_path=None,
                  callbacks=None, build_index_on_disk=True,
                  neighbour_matrix=None, verbose=1):
-
         self.embedding_dims = embedding_dims
         self.k = k
         self.distance = distance
         self.batch_size = batch_size
         self.epochs = epochs
         self.n_epochs_without_progress = n_epochs_without_progress
         self.knn_distance_metric = knn_distance_metric
-        self.ntrees = ntrees
+        self.n_trees = n_trees
```

Comment:
This could work, along with some docstring updates and a warning. I will work on it.

Reply:
I made some changes to schedule `ntrees` for deprecation in favor of `n_trees`. `ntrees` was added back to `Ivis.__init__()` and its default value, unlike `n_trees`, was set to `None`. Whenever `ntrees != None` (i.e., the user set a value for it), a `FutureWarning` is displayed when the instance is fit to data, and the value of `ntrees` is passed to `n_trees`. This action is performed every time `Ivis.fit()` is called, as this behavior is in `Ivis._validate_parameters()`, which is called by `Ivis.fit()`.

This solution is still not ideal, as it is not recommended to change the input hyper-parameters during execution. However, since the user is warned every time, the validation logic in a place other than `Ivis.__init__()`, and this behavior is temporary, I think this might be acceptable. What do you think?

Reply:
Yes, I agree that this is fine for temporary behavior. Should be clear that users should to move to `n_trees`.